File: app/src/main/java/wannabit/io/cosmostaion/dialog/StarnameDomainDialog.java
package wannabit.io.cosmostaion.dialog;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import wannabit.io.cosmostaion.R;

public class StarnameDomainDialog extends DialogFragment {

    private RecyclerView mRecyclerView;
    private TextView mDialogTitle;
    private DomainListAdapter mDomainListAdapter;
    private ArrayList<String> mStarnameDomain = new ArrayList<>();

    public static StarnameDomainDialog newInstance(Bundle bundle) {
        StarnameDomainDialog frag = new StarnameDomainDialog();
        frag.setArguments(bundle);
        return frag;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        getDialog().getWindow().setBackgroundDrawableResource(R.drawable.layout_trans_with_border);
        View view = LayoutInflater.from(getActivity()).inflate(R.layout.dialog_template_recycler, null);
        mDialogTitle = view.findViewById(R.id.dialog_title);
        mRecyclerView = view.findViewById(R.id.recycler);
        mStarnameDomain = getArguments().getStringArrayList("domain");

        mDialogTitle.setText(R.string.str_select_starname_domain);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
        mRecyclerView.setHasFixedSize(true);
        mDomainListAdapter = new DomainListAdapter();
        mRecyclerView.setAdapter(mDomainListAdapter);
        return view;
    }

    private class DomainListAdapter extends RecyclerView.Adapter<DomainListAdapter.DomainListHolder> {

        @NonNull
        @Override
        public DomainListHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
            return new DomainListHolder(getLayoutInflater().inflate(R.layout.item_dialog_starname_domain_list, viewGroup, false));
        }

        @Override
        public void onBindViewHolder(@NonNull DomainListHolder holder, int position) {
            onBindDomainItemViewHolder(holder, position);
        }

        private void onBindDomainItemViewHolder(DomainListHolder holder, int position) {
            final String domain = mStarnameDomain.get(position);
            holder.domainName.setText(domain);
            holder.rootLayer.setOnClickListener(v -> {
                Intent resultIntent = new Intent();
                resultIntent.putExtra("position", position);
                getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, resultIntent);
                getDialog().dismiss();
            });
        }

        @Override
        public int getItemCount() {
            return mStarnameDomain.size();
        }

        public class DomainListHolder extends RecyclerView.ViewHolder {
            LinearLayout rootLayer;
            TextView domainName;

            public DomainListHolder(@NonNull View itemView) {
                super(itemView);
                rootLayer = itemView.findViewById(R.id.rootLayer);
                domainName = itemView.findViewById(R.id.domainName);
            }
        }

    }
}

File: app/src/main/java/wannabit/io/cosmostaion/dialog/StarNameDomainDialog.kt
package wannabit.io.cosmostaion.dialog

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import wannabit.io.cosmostaion.R
import wannabit.io.cosmostaion.base.BaseConstant
import wannabit.io.cosmostaion.databinding.DialogTemplateRecyclerBinding
import wannabit.io.cosmostaion.databinding.ItemDialogStarnameDomainListBinding
import wannabit.io.cosmostaion.dialog.StarNameDomainDialog.DomainListAdapter.DomainListHolder

class StarNameDomainDialog : DialogFragment() {
    private var dialogTemplateRecyclerBinding: DialogTemplateRecyclerBinding? = null
    private var mStarNameDomain: ArrayList<String>? = ArrayList()
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        dialog!!.window!!.setBackgroundDrawableResource(R.drawable.layout_trans_with_border)
        dialogTemplateRecyclerBinding =
            DialogTemplateRecyclerBinding.inflate(inflater, container, false)
        mStarNameDomain = arguments!!.getStringArrayList("domain")
        dialogTemplateRecyclerBinding!!.dialogTitle.setText(R.string.str_select_starname_domain)
        dialogTemplateRecyclerBinding!!.recycler.layoutManager =
            LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        dialogTemplateRecyclerBinding!!.recycler.setHasFixedSize(true)
        dialogTemplateRecyclerBinding!!.recycler.adapter = DomainListAdapter()
        return dialogTemplateRecyclerBinding!!.root
    }

    private inner class DomainListAdapter : RecyclerView.Adapter<DomainListHolder>() {
        override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): DomainListHolder {
            return DomainListHolder(
                ItemDialogStarnameDomainListBinding.inflate(
                    layoutInflater
                )
            )
        }

        override fun onBindViewHolder(holder: DomainListHolder, position: Int) {
            onBindDomainItemViewHolder(holder, position)
        }

        private fun onBindDomainItemViewHolder(holder: DomainListHolder, position: Int) {
            val domain = mStarNameDomain!![position]
            holder.itemDialogStarnameDomainListBinding.domainName.text = domain
            holder.itemDialogStarnameDomainListBinding.rootLayer.setOnClickListener {
                val result = Bundle()
                result.putInt(BaseConstant.POSITION, position)
                parentFragmentManager.setFragmentResult(STAR_NAME_DOMAIN_BUNDLE_KEY, result)
                dismiss()
            }
        }

        override fun getItemCount(): Int {
            return mStarNameDomain!!.size
        }

        inner class DomainListHolder(val itemDialogStarnameDomainListBinding: ItemDialogStarnameDomainListBinding) :
            RecyclerView.ViewHolder(
                itemDialogStarnameDomainListBinding.root
            )
    }

    override fun onDestroyView() {
        super.onDestroyView()
        dialogTemplateRecyclerBinding = null
    }

    companion object {
        const val STAR_NAME_DOMAIN_BUNDLE_KEY = "starNameDomain"

        @JvmStatic
        fun newInstance(bundle: Bundle?): StarNameDomainDialog {
            val frag = StarNameDomainDialog()
            frag.arguments = bundle
            return frag
        }
    }
}